src/textbuf.cpp

```diff
@@ -59,7 +59,7 @@ bool Textbuf::DeleteChar(uint16_t keycode)
 
 	if (!CanDelChar(backspace)) return false;
 
-	char *s = this->buf + this->caretpos;
+	char *s = &this->buf[this->caretpos];
```

Comment:
Does it make sense/take much effort to have this (and similar later on) as proper string iterators instead of pointers to the string's data?

Reply:
It makes sense, though they'll quickly get hung up on the `Utf8` functions (e.g. `Utf8PrevChar` and `Utf8Decode`) not working for string::iterators yet.
It would also mean changing `GetCharPosInString` to work with iterators.

Reply:
Well, +21,-10 for the ones that are not `GetCharPosInString`.